File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/governance/GovernanceItemsAtTest.java
package opensdk.sdk.apis.governance;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.GovernanceItemsAtResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

@DisplayName("Governance RPC Test")
public class GovernanceItemsAtTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC governance_itemsAt")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        int blockTag = 0;

        GovernanceItemsAtResponse response = sdk.governance.itemsAt(blockTag).send();
        response.getResult();
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/governance/GovernanceItemsAt.test.js
const OpenSdk = require("@klaytn/web3rpc");
const { expect } = require("@jest/globals");
const { RPC } = require("../constant");

const sdk = new OpenSdk.GovernanceApi(new OpenSdk.ApiClient(RPC));

describe('governance_itemsAt API', () => {
    test('should return governance_itemsAt', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined();
            expect(data['governance.governingnode']).toMatch(/^0x.*$/gm);
            done();
        };

        const blockNumber = 89;

        sdk.itemsAt(blockNumber, {}, callbackOne);
    });
});

